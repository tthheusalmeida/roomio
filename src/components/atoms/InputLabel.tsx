import { mergeClassNames } from "@/utils/classNames";
import { ReactNode } from "react";

export default function InputLabel({
  text,
  children,
  className,
  maxInput,
  currentInput,
}: {
  text: string;
  children: ReactNode;
  className?: string;
  maxInput?: number;
  currentInput?: string;
}) {
  return (
    <label className={mergeClassNames("my-4 block", className)}>
      <div className="w-full flex justify-between items-baseline">
        <span className="block font-semibold my-2">{text}</span>
        <span
          className={mergeClassNames(
            "block font-medium text-xs my-2 transition-all duration-200",
            maxInput && Number(currentInput?.length) > maxInput
              ? "text-red-500 font-bold"
              : ""
          )}
        >
          {currentInput?.length}/{maxInput}
        </span>
      </div>
      {children}
    </label>
  );
}
