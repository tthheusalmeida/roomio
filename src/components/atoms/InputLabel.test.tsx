import { render, screen } from "@testing-library/react";
import InputLabel from "./InputLabel";

describe("InputLabel component", () => {
  it("renders text and children correctly", () => {
    const { asFragment } = render(
      <InputLabel text="Username">
        <input />
      </InputLabel>
    );

    expect(screen.getByText("Username")).toBeInTheDocument();

    expect(screen.getByRole("textbox")).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders character counter correctly", () => {
    const { asFragment } = render(
      <InputLabel text="Bio" currentInput="Hello" maxInput={10}>
        <input />
      </InputLabel>
    );

    expect(screen.getByText("5/10")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("applies error classes when input exceeds maxInput", () => {
    render(
      <InputLabel text="Bio" currentInput="Hello World" maxInput={5}>
        <input />
      </InputLabel>
    );

    const counter = screen.getByText("11/5");
    expect(counter).toHaveClass("text-red-500", "font-bold");
  });

  it("applies extra classes passed via className", () => {
    const { asFragment } = render(
      <InputLabel text="Name" className="extra-class">
        <input />
      </InputLabel>
    );

    const label = screen.getByText("Name").closest("label");
    expect(label).toHaveClass("my-4", "block", "extra-class");

    expect(asFragment()).toMatchSnapshot();
  });
});
