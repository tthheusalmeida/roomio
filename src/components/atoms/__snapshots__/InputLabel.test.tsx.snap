// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`InputLabel component applies extra classes passed via className 1`] = `
<DocumentFragment>
  <label
    class="my-4 block extra-class"
  >
    <div
      class="w-full flex justify-between items-baseline"
    >
      <span
        class="block font-semibold my-2"
      >
        Name
      </span>
      <span
        class="block font-medium text-xs my-2 transition-all duration-200"
      >
        /
      </span>
    </div>
    <input />
  </label>
</DocumentFragment>
`;

exports[`InputLabel component renders character counter correctly 1`] = `
<DocumentFragment>
  <label
    class="my-4 block"
  >
    <div
      class="w-full flex justify-between items-baseline"
    >
      <span
        class="block font-semibold my-2"
      >
        Bio
      </span>
      <span
        class="block font-medium text-xs my-2 transition-all duration-200"
      >
        5/10
      </span>
    </div>
    <input />
  </label>
</DocumentFragment>
`;

exports[`InputLabel component renders text and children correctly 1`] = `
<DocumentFragment>
  <label
    class="my-4 block"
  >
    <div
      class="w-full flex justify-between items-baseline"
    >
      <span
        class="block font-semibold my-2"
      >
        Username
      </span>
      <span
        class="block font-medium text-xs my-2 transition-all duration-200"
      >
        /
      </span>
    </div>
    <input />
  </label>
</DocumentFragment>
`;
